microsoft-edge/web-platform/ad-selection-api.md

```diff
@@ -0,0 +1,255 @@
+
+Images for seller services:
+* [SellerFrontEnd service](https://mcr.microsoft.com/product/ad-selection/azure/seller-frontend-service)
+* [Auction service](https://mcr.microsoft.com/product/ad-selection/azure/auction-service)
+
+Images for buyer services:
+* [BuyerFrontEnd service](https://mcr.microsoft.com/product/ad-selection/azure/buyer-frontend-service)
+* [Bidding service](https://mcr.microsoft.com/product/ad-selection/azure/bidding-service)
+* [k-anonymity service](https://mcr.microsoft.com/product/ad-selection/azure/k-anonymity-service)
+
+
+<!-- ------------------------------ -->
+#### Services for sellers
+
+| Service | Description |
+| --- | --- |
+| **SellerFrontEnd** | Provides a `/SelectAd` gRPC endpoint, which receives requests from the seller's ad service<!-- todo: what's the bold CamelCase name of that service? --> to initiate the Protected Audience auction flow. |
+| **Auction** | Provides a `/ScoreAds` gRPC endpoint, which receives requests from the **SellerFrontEnd** service, containing bids that are participating in the auction.  Responds with a score value that the **SellerFrontEnd** service uses to choose the winner. |
+| **Key/Value** | Receives requests from the **SellerFrontEnd** service, which contain lookup keys from buyers' bids (such as `ad_render_urls`).  Returns real-time scoring signals that are required for the auction.  Runs in Bring Your Own Service (BYOS) mode, so the seller does not need to deploy this service in a trusted execution environment (TEE), and can instead use the image that's provided by Microsoft. |
+
+
+<!-- ------------------------------ -->
+#### Services for buyers
+
+| Service | Description |
+| --- | --- |
+| **BuyerFrontEnd** | Provides a `/GetBids` gRPC endpoint, which receives requests from the **SellerFrontEnd** service to initiate the bidding flow. |
+| **Bidding** | Provides a `/GenerateBids` endpoint, which receives requests from the **BuyerFrontEnd** service to handle the bidding and generate a bid.  Generates a bid, chooses the winner, and selects the banner to be rendered. |
+| **Key/Value** | Receives requests from the **BuyerFrontEnd** service and returns real-time buyer data required for bidding, corresponding to lookup keys from Interest Groups.  Such a request happens once per workflow.  Runs in Bring Your Own Service (BYOS) mode. |
+| **Bidding Selection & Key/Value** | Receives requests from the **Bidding** service to select and return additional ad banners (candidates) that can participate in bidding.  The **Bidding**<!-- todo: which service? --> service can also return additional signals that are needed for bidding.  The **Bidding** service may send multiple requests to the **Bidding Selection & Key/Value** service, or may choose not to send any requests, given that the **Bidding Selection & Key/Value** service<!-- todo: which service? --> is optional.  The **Bidding Selection & Key/Value** service must be deployed in a trusted execution environment (TEE). |
```

Comment:
"The service can also return additional signals that are needed for bidding."  Which service, b/c near here, discussing two services.  draft Doc:
![image](https://github.com/user-attachments/assets/20e87557-6d54-4c31-9d6c-1dbce431e7a1)

Reply:
1) "The **Bidding Selection & Key/Value** can also return additional signals needed for bidding.

2) "The Bidding Service may send multiple requests to the Selection & Key/Value service or may choose not to send any, as the **Bidding Selection & Key/Value** is optional.

Reply:
result:

The **Bidding Selection & Key/Value** service can also return additional signals that are needed for bidding.

The **Bidding** service may send multiple requests to the **Bidding Selection & Key/Value** service, or may choose not to send any requests, because the **Bidding Selection & Key/Value** service is optional.